### ISSUE TITLE
Check all valid import forms during import tracking

### ISSUE THREAD

**Issue by charliermarsh** [charliermarsh/ruff#713]
Assume that we're trying to import `typing.re.Match` (and, also, that `re.Match` doesn't exist, just for sake of example).

All of these forms (marked `# OK` need to work ):

```py
# OK
from typing import re

print(re.Match)

# Error
from typing import *

print(re.Match)

# OK
from typing.re import *

print(Match)

# OK
from typing.re import Match

print(Match)

# OK
import typing

print(typing.re.Match)

# OK
import typing.re

print(typing.re.Match)
```

In particular, we don't catch this:

```py
from typing import re

re.Match["foo"]
```
